### DISCUSSION TITLE
Windows support

### DISCUSSION THREAD

**Discussion by thbeu** [jenstroeger/python-package-template#381] [Q&A]
I have difficulties to setup and evaluate the project template on Windows as there is no `make` available - neither in Git bash nor natively in PowerShell. As such, I am not able to run `make requirements` or `make sbom`. What is the recommended usage to get started on Windows?

**Comment by jenstroeger** [comment]
Hello @thbeu, thank you for your questions!

We run and test the template on Github Windows runners ([docs](https://github.com/actions/runner-images/blob/main/images/win/Windows2022-Readme.md)) which already have [CMake](https://cmake.org/) installed. I think you have two options:

- Download and install CMake from [here](https://cmake.org/download/#latest); or
- Install the [Windows Subsystem for Linux](https://learn.microsoft.com/en-us/windows/wsl/) which essentially is a modified Ubuntu distro with access to all of the Linux tooling.

I’d suggest to try CMake first!

**Comment by thbeu** [comment]
I have CMake installed but there is no make tool provided.
I also have WSL setup but do not think the venv will work together with VS code.

**Comment by jenstroeger** [comment]
> I have CMake installed but there is no make tool provided.

Right, and it wouldn’t be of much use anyway considering our Makefile which [was built for GNU Make and requires `bash`](https://github.com/jenstroeger/python-package-template/blob/7bf474b781efa1fb0e69dcf31f968675695f10bc/Makefile#L2-L4). Looking at the Github Windows runner, it does have heaps of additional tools installed and I’m not sure if you’d like to add them to your setup.

I do have a Windows VM here, and can try to figure out a minimal set of tools you need to install on Windows for our package. Otherwise, your best bet is…

> I also have WSL setup but do not think the venv will work together with VS code.

You think it won’t work because of discussion https://github.com/jenstroeger/python-package-template/discussions/382?

**Comment by jenstroeger** [comment]
I’ve not had a chance to look into this. Do you still face the `make` problems?
